Docs and formatting enforcement

ches opened the pull request:
Hiya!

I started hacking on the plugin, and I noticed that I was messing up the indentation convention because my default settings for VimL are different—the VimL code on this project is indented with tabs, the doc file was mostly spaces with tabs mixed in some places. To prevent myself (and other possible contributors) from mixing up the indentation, I added modelines that best match the existing code: I kept tabs in the VimL code and set `'noexpandtab'`, and in the doc set `'expandtab'` and converted any places that had tabs to use spaces.

The `'tabstop'` value is a matter of personal taste, so you might change it if you prefer something else. Vim suggests keeping the value at 8 because that's how tab-indented code will usually be displayed by tools others than Vim, but personally looking at 8-column wide indents while working in Vim drives me crazy so I chose 4 as hopefully a compromise (some people think 2 is too small...).

While I was reformatting the doc file, I tried to clarify wording in parts of the docs. I hope this is okay.

I started this to add a feature or two that I'd like, but now I'm not sure if they're going to work without substantial changes to the plugin. I'm still going to try and I'll send pull requests if I get things working without a major rewrite, but I wanted to go ahead and send these changes before I forgot them.

yssl commented:
Hi ches!
I really appreciate your changes, especially for the wording!
I've leave some comments on the diff, so please check it.

ches commented:
Okay! Changed the indentation on the doc file contents section to 8 spaces, squashed that into the original commit. Let me know if there's anything else :smile:

yssl commented:
Thank you so much for your contribution!
